lib/Makefile.am

```diff
@@ -8,5 +8,9 @@ libsysfs_la_SOURCES = sysfs_utils.c sysfs_attr.c sysfs_class.c dlist.c \
       sysfs_device.c sysfs_driver.c sysfs_bus.c sysfs_module.c sysfs.h
 libsysfs_la_CPPFLAGS = -I$(top_srcdir)/include
 libsysfs_la_LDFLAGS = -version-info 2:1:0
+if HAVE_LINKER_VERSION_SCRIPT
+libsysfs_la_LDFLAGS += -Wl,--version-script=$(srcdir)/libsysfs.map
+endif
+EXTRA_libsysfs_la_DEPENDENCIES = libsysfs.map
```

Comment:
Thanks for the improvements. I might be completely wrong here, should the `EXTRA_ibsysfs_la_DEPENDENCIES` moved into above if block? So that the dependency is only valid of the `HAVE_LINKER_VERSION_SCRIPT` is true

Reply:
Ah nice catch, yes. While I don't think having the unconditional dependency hurts much as the file is included in the repo and  distribution, it is indeed strictly speaking not correct. :)